shadowsocks/tcp.go

```diff
@@ -115,15 +102,25 @@ func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, o
 
 type tcpService struct {
 	listener    *net.TCPListener
-	ciphers     *CipherList
+	ciphers     CipherList
```

Comment:
The CipherList needs to be a pointer for both TCP and UDP so we can updated it on config change without restarting the services.

It's set here: https://github.com/Jigsaw-Code/outline-ss-server/blob/76edca1e44933857f2cefefe1ed9be58940d100a/server.go#L150

Reply:
Thanks for the explanation.  I've reverted this change.

Reply:
Maybe we should add a comment explaining why it has to be a pointer, so readers know.